python-avd/pyavd/_eos_designs/structured_config/network_services/router_ospf.py

```diff
@@ -61,15 +61,25 @@ def router_ospf(self: AvdStructuredConfigNetworkServices) -> dict | None:
                     msg = f"'ospf.process_id' or 'vrf_id' under vrf '{vrf['name']}"
                     raise AristaAvdInvalidInputsError(msg)
 
-                process = {
-                    "id": process_id,
-                    "vrf": vrf["name"] if vrf["name"] != "default" else None,
-                    "passive_interface_default": True,
-                    "router_id": default(get(vrf, "ospf.router_id"), self.shared_utils.router_id),
-                    "no_passive_interfaces": ospf_interfaces,
-                    "bfd_enable": get(vrf, "ospf.bfd"),
-                    "max_lsa": get(vrf, "ospf.max_lsa"),
-                }
+                process = {}
+
+                process.update(
+                    {
+                        "id": process_id,
+                        "vrf": vrf["name"] if vrf["name"] != "default" else None,
+                        "passive_interface_default": True,
+                    }
+                )
+                if self.shared_utils.use_router_general_for_router_id is False:
+                    process["router_id"] = default(get(vrf, "ospf.router_id"), self.shared_utils.router_id)
```

Comment:
If the router_id is set specifically under ospf.router_id for this VRF, we should honor that. So the order of precedence needs to change here.
First take ospf.router_id. If not set, check for use_router_general_... and don't print anything if true. Finally use self.shared_utils.router_id.

Reply:
Fixed.